### PR TITLE
Adding the ability to handle APIs within the app

### PR DESCRIPTION
**Pull request by nathanbillis**
This PR will add the ability to call APIs and receive the data into the application for using in other methods.

### CODE REVIEW

**Comment by nathanbillis** [comment]
image

**Comment by profjb58** [comment]
Nearly everything seems ok. 
Never implemented an API before but the only issues I found where that the assertNotNull for the last 2 tests isn't of much use since the application is saying the value of an integer can never be null. 

Also i'm assuming 200 means success but I got a 405 error on the API caller with data method.

**Comment by profjb58** [comment]
image
image

**Comment by profjb58** [comment]
If the GET and POST methods haven't yet been added could just change to check the value of the integer returned is within the valid HTTPS returns so 100 - 511

**Comment by nathanbillis** [comment]
> Nearly everything seems ok.
> Never implemented an API before but the only issues I found where that the assertNotNull for the last 2 tests isn't of much use since the application is saying the value of an integer can never be null.
> 
> Also i'm assuming 200 means success but I got a 405 error on the API caller with data method.

Interesting... The last two are assertNotNull because there's no functionality yet to read in JSONObjects

File: app/src/test/java/com/group4sweng/scranplan/APIHandler/APICallerTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        int i = new APICaller("http://www.google.com/search?").withData("q=google").send();
+        assertNotNull(i);
+    }
+
+    @Test
+    public void CallWithHeader() throws Exception{
+        int i = new APICaller("http://www.google.com/search?").withHeaders("q:header").send();
+        assertNotNull(i);
+    }
+
```
Comment (profjb58) [inline app/src/test/java/com/group4sweng/scranplan/APIHandler/APICallerTest.java]: AssertNotNull false check.
could do...

```
if(i >= 100 && i <= 511){
   fail("message")
}
```

File: app/src/test/java/com/group4sweng/scranplan/APIHandler/APICallerTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    public void MakeJSONRequest() throws Exception {
+        JSONObject object = new APICaller("http://www.google.com").sendAndReadJSON();
+        assertNotNull(object);
+    }
+
+    @Test
+    public void CallWithData() throws Exception{
+        int i = new APICaller("http://www.google.com/search?").withData("q=google").send();
+        assertNotNull(i);
+    }
```
Comment (profjb58) [inline app/src/test/java/com/group4sweng/scranplan/APIHandler/APICallerTest.java]: 405 return.

Comment (nathanbillis) [reply app/src/test/java/com/group4sweng/scranplan/APIHandler/APICallerTest.java]: Ta

File: app/src/test/java/com/group4sweng/scranplan/APIHandler/APICallerTest.java (hunk 1 above)
Comment (profjb58) [reply app/src/test/java/com/group4sweng/scranplan/APIHandler/APICallerTest.java]: Just realized it should be if(i< 100 || i > 511) :P

File: app/src/test/java/com/group4sweng/scranplan/APIHandler/APICallerTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Test
+    public void SendFakeUrl() throws Exception{
+        // make a non url request
+        int i = new APICaller("notaurl").send();
+        assertEquals(-1, i);
+    }
+
+    @Test
+    public void MakeJSONRequest() throws Exception {
```
Comment (nathanbillis) [inline app/src/test/java/com/group4sweng/scranplan/APIHandler/APICallerTest.java]: Need to also test data and json request

**Comment by nathanbillis** [comment]
Thanks @profjb58 this has been helpful, I'm going to close and re-open when I've fixed this :)

**Pull request closed without merging**